apps/emqx_management/src/emqx_mgmt_api_configs.erl

```diff
@@ -16,97 +16,106 @@
 
 -module(emqx_mgmt_api_configs).
 
+-include_lib("hocon/include/hoconsc.hrl").
 -behaviour(minirest_api).
 
--import(emqx_mgmt_util, [ schema/1
-                        , schema/2
-                        , error_schema/2
-                        ]).
-
 -export([api_spec/0]).
+-export([paths/0, schema/1, fields/1]).
 
--export([ config/3
-        , config_reset/3
-        ]).
+-export([config/3, config_reset/3, configs/3, get_full_config/0]).
 
 -export([get_conf_schema/2, gen_schema/1]).
 
--define(PARAM_CONF_PATH, [#{
-    name => conf_path,
-    in => query,
-    description => <<"The config path separated by '.' character">>,
-    required => false,
-    schema => #{type => string, default => <<".">>}
-}]).
-
--define(PREFIX, "/configs").
--define(PREFIX_RESET, "/configs_reset").
-
--define(MAX_DEPTH, 1).
-
+-define(PREFIX, "/configs/").
+-define(PREFIX_RESET, "/configs_reset/").
 -define(ERR_MSG(MSG), list_to_binary(io_lib:format("~p", [MSG]))).
-
--define(CORE_CONFS, [
-    %% from emqx_machine_schema
-    log, rpc,
-    %% from emqx_schema
-    zones, mqtt, flapping_detect, force_shutdown, force_gc, conn_congestion, rate_limit, quota,
-    broker, alarm, sysmon,
-    %% from other apps
-    emqx_dashboard, emqx_management]).
+-define(EXCLUDES, [listeners, node, cluster, gateway, rule_engine]).
 
```

Comment:
read from the field will miss a lot of configurations.
so we get configuration from emqx_machine_schema:roots().
and exclude some no-hot update fields, such `node`.